src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt

```diff
@@ -70,9 +74,61 @@ class CourtSentencingSynchronisationService(
     }
   }
 
+  suspend fun nomisCourtAppearanceInserted(event: CourtAppearanceEvent) {
+    val telemetry =
+      mutableMapOf(
+        "nomisCourtAppearanceId" to event.courtAppearanceId.toString(),
+        "offenderNo" to event.offenderIdDisplay,
+        "nomisBookingId" to event.bookingId.toString(),
+      )
+    if (event.auditModuleName == "DPS_SYNCHRONISATION") {
+      telemetryClient.trackEvent("court-appearance-synchronisation-created-skipped", telemetry)
+    } else {
+      val nomisCourtAppearance =
+        nomisApiService.getCourtAppearance(
+          offenderNo = event.offenderIdDisplay,
+          courtAppearanceId = event.courtAppearanceId,
+        )
+      mappingApiService.getCourtAppearanceOrNullByNomisId(event.courtAppearanceId)?.let { mapping ->
+        telemetryClient.trackEvent(
+          "court-appearance-synchronisation-created-ignored",
+          telemetry + ("dpsCourtAppearanceId" to mapping.dpsCourtAppearanceId),
+        )
+      } ?: let {
+        val dpsCaseId =
+          nomisCourtAppearance.caseId?.let { mappingApiService.getCourtCaseOrNullByNomisId(it)?.dpsCourtCaseId }
```

Comment:
need to look up the court case mapping also as it is required for the appearance create

Reply:
see  below